zhaquirks/tuya/thermostat_88teujp.py

```diff
@@ -0,0 +1,251 @@
+        if value == 1:  # alert
+            self._update_attribute(
+                BATTERY_PERCENTAGE_REMAINING_ATTR, 0
+            )  # report 0% battery
+        else:
+            self._update_attribute(
+                BATTERY_PERCENTAGE_REMAINING_ATTR, 200
+            )  # report 100% battery
+
+
+class ThermostatCluster(LocalDataCluster, Thermostat):
+    """Thermostat cluster."""
+
+    cluster_id = Thermostat.cluster_id
+
+    _CONSTANT_ATTRIBUTES = {
+        MIN_HEAT_SETPOINT_ATTR: 500,
+        MAX_HEAT_SETPOINT_ATTR: 3000,
+        CTRL_SEQ_OF_OPER_ATTR: Thermostat.ControlSequenceOfOperation.Heating_Only,  # the device supports heating mode
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.thermostat_bus.add_listener(self)
+
+    def occupied_heating_setpoint_reported(self, value):
+        """Handle reported occupied heating setpoint state."""
+        self._update_attribute(OCCUPIED_HEATING_SETPOINT_ATTR, value * 10)
+        _LOGGER.debug("reported occupied heating setpoint: %d", value)
+
+    def local_temp_reported(self, value):
+        """Handle reported local temperature."""
+        self._update_attribute(LOCAL_TEMP_ATTR, value * 10)
+        _LOGGER.debug("reported local temperature: %d", value)
+
+    def system_mode_reported(self, value):
+        """Handle reported system mode."""
+        if value == 1:
+            self._update_attribute(SYSTEM_MODE_ATTR, Thermostat.SystemMode.Heat)
+            self._update_attribute(RUNNING_MODE_ATTR, Thermostat.RunningMode.Heat)
+            _LOGGER.debug("reported system_mode: heat")
+        else:
+            self._update_attribute(SYSTEM_MODE_ATTR, Thermostat.SystemMode.Off)
+            self._update_attribute(RUNNING_MODE_ATTR, Thermostat.RunningMode.Off)
+            _LOGGER.debug("reported system_mode: off")
+
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Override remote writes."""
+        if "system_mode" in attributes:
+            mode = attributes.get("system_mode")
+
+            if mode == Thermostat.SystemMode.Off:
+                await self.send_tuya_command(SYSTEM_MODE_COMMAND_ID, [1, 0])
+                _LOGGER.debug("set system_mode: off")
+            if mode == Thermostat.SystemMode.Heat:
+                await self.send_tuya_command(SYSTEM_MODE_COMMAND_ID, [1, 1])
+                _LOGGER.debug("set system_mode: heat")
+        elif "occupied_heating_setpoint" in attributes:
+            temperature = int(attributes.get("occupied_heating_setpoint") / 10)
+            await self.send_tuya_command(
+                OCCUPIED_HEATING_SETPOINT_COMMAND_ID, decimal_to_payload(temperature)
+            )
+            _LOGGER.debug("set occupied_heating_setpoint: %d", temperature)
+        return (foundation.Status.SUCCESS,)
+
+    async def send_tuya_command(self, cmd, data):
```

Comment:
This code already exists in `__init__.py`. You should reuse it, even if you decide not to join efforts with me in #538.